scrapy/utils/reqser.py

```diff
@@ -75,7 +79,16 @@ def _find_method(obj, func):
             pass
         else:
             if func_self is obj:
-                return six.get_method_function(func).__name__
+                name = six.get_method_function(func).__name__
+                if _is_private_method(name):
+                    qualname = getattr(func, '__qualname__', None)
+                    if qualname is None:
+                        classname = obj.__class__.__name__.lstrip('_')
+                        name = '_%s%s' % (classname, name)
+                    else:
+                        splits = qualname.split('.')
+                        name = '_%s%s' % (splits[-2], splits[-1])
```

Comment:
do you think it is possible to extract this logic to a separate function, e.g. `_mangled_name` (?), to make the intention of the code more clear, and also to be able to add tests for it? You can check that names produced are the same as Python makes.

Reply:
I added tests for the two cases (own method, inherited method)... I feel like further testing would be going into "writing tests to prove Python behavior" rather than "writing tests to prove Scrapy behavior" but please let me know if you think a test with a nested class or something could be helpful.

Reply:
👍 thanks @andrewbaxter!